be/src/runtime/sender_queue.cpp

```diff
@@ -574,17 +575,27 @@ void DataStreamRecvr::PipelineSenderQueue::clean_buffer_queues() {
 }
 
 Status DataStreamRecvr::PipelineSenderQueue::try_to_build_chunk_meta(const PTransmitChunkParams& request) {
-    ScopedTimer<MonotonicStopWatch> wait_timer(_recvr->_sender_wait_lock_timer);
-    std::lock_guard<Mutex> l(_lock);
-    wait_timer.stop();
     // We only need to build chunk meta on first chunk and not use_pass_through
     // By using pass through, chunks are transmitted in shared memory without ser/deser
     // So there is no need to build chunk meta.
-    if (_chunk_meta.types.empty() && !request.use_pass_through()) {
-        SCOPED_TIMER(_recvr->_deserialize_chunk_timer);
-        auto& pchunk = request.chunks(0);
-        return _build_chunk_meta(pchunk);
+    if (request.use_pass_through()) {
+        return Status::OK();
+    }
+    if (_is_chunk_meta_built) {
+        return Status::OK();
     }
+
+    ScopedTimer<MonotonicStopWatch> wait_timer(_recvr->_sender_wait_lock_timer);
```

Comment:
can be used concurrently?

Reply:
> can be used concurrently?

of course yes, _sender_wait_lock_timer is maintained by a atomic variable